server/src/main/java/org/eclipse/lsp/cobol/service/ConfigurationImpl.java

```diff
@@ -0,0 +1,104 @@
+/*
+ * Copyright (c) 2022 Broadcom.
+ * The term "Broadcom" refers to Broadcom Inc. and/or its subsidiaries.
+ *
+ * This program and the accompanying materials are made
+ * available under the terms of the Eclipse Public License 2.0
+ * which is available at https://www.eclipse.org/legal/epl-2.0/
+ *
+ * SPDX-License-Identifier: EPL-2.0
+ *
+ * Contributors:
+ *    Broadcom, Inc. - initial API and implementation
+ *
+ */
+package org.eclipse.lsp.cobol.service;
+
+import com.google.common.collect.ImmutableList;
+import com.google.common.collect.Streams;
+import com.google.gson.JsonArray;
+import com.google.gson.JsonElement;
+import com.google.inject.Singleton;
+import lombok.extern.slf4j.Slf4j;
+import org.eclipse.lsp.cobol.core.model.tree.EmbeddedCodeNode;
+
+import javax.inject.Inject;
+import java.util.*;
+import java.util.concurrent.CompletableFuture;
+
+import static java.util.Collections.unmodifiableList;
+import static java.util.Collections.unmodifiableSet;
+import static java.util.stream.Collectors.toList;
+import static org.eclipse.lsp.cobol.service.utils.SettingsParametersEnum.*;
+
+/** This service fetches configuration settings from the client. */
+@Slf4j
+@Singleton
+public class ConfigurationImpl implements Configuration {
+  SQLBackend sqlBackend;
+  Set<EmbeddedCodeNode.Language> features = new HashSet<>();
+  List<String> flavors = ImmutableList.of();
+
+  private SettingsService settingsService;
+
+  @Inject
+  public ConfigurationImpl(SettingsService settingsService) {
+    this.settingsService = settingsService;
+  }
+
+  @Override
+  public CompletableFuture<Void> getConfigurationFromSettings() {
+    return settingsService
+            .getConfigurations(
+                    Arrays.asList(TARGET_SQL_BACKEND.label, ANALYSIS_FEATURES.label, FLAVORS.label))
+            .thenAccept(this::updateConfigs);
```

Comment:
Maybe it will be better to handle the Future synchronously? Because right now in CobolLanguageServer:
```
  public void initialized(@Nullable InitializedParams params) {
     <....>
    configuration.getConfigurationFromSettings();
  }
```
Here we can assume that settings were updated when the `initialized` method was done, but they are not.